test/test_centroids.py

```diff
@@ -1,168 +1,145 @@
 import os
-from unittest import TestCase
 import numpy as np
 import numpy.testing as nt
 import uxarray as ux
 from pathlib import Path
-from uxarray.grid.coordinates import _populate_face_centroids, _populate_edge_centroids, _populate_face_centerpoints, _is_inside_circle, _circle_from_three_points, _circle_from_two_points
-from uxarray.grid.coordinates import _normalize_xyz
+from uxarray.grid.coordinates import (
+    _populate_face_centroids,
+    _populate_edge_centroids,
+    _populate_face_centerpoints,
+    _is_inside_circle,
+    _circle_from_three_points,
+    _circle_from_two_points,
+    _normalize_xyz,
+)
 
 current_path = Path(os.path.dirname(os.path.realpath(__file__)))
 
 gridfile_CSne8 = current_path / "meshfiles" / "scrip" / "outCSne8" / "outCSne8.nc"
 mpasfile_QU = current_path / "meshfiles" / "mpas" / "QU" / "mesh.QU.1920km.151026.nc"
 
+def test_centroids_from_mean_verts_triangle():
+    """Test finding the centroid of a triangle."""
+    test_triangle = np.array([(0, 0, 1), (0, 0, -1), (1, 0, 0)])
+    expected_centroid = np.mean(test_triangle, axis=0)
+    norm_x, norm_y, norm_z = _normalize_xyz(
+        expected_centroid[0], expected_centroid[1], expected_centroid[2]
+    )
 
-class TestCentroids(TestCase):
```

Comment:
Test cases in this script were grouped into two different classes before. While I don't know the reason for that distinction, I am hesitant that this change here remove the distinction. You'd still be able to define the same classes with `pytest`, why not?

This question may apply to all the other scripts in the same situation.

Reply:
We currently have:


```
class ABC(TestCase):
    def test1(self):
     def test2(self):
class XYZ(TestCase):
    def test1(self):

```
Better to move to this structure as a complete pytest way as it aligns better with pytest's focus on functions and fixtures. This is harder, but more long term and maintainable. 

```
def test_abc_1(setup_data):
    # .. your test logic from ABC.test1 ...

def test_abc_2(setup_data):
    # .. your test logic from ABC.test2 ...

def test_xyz_1(setup_data):
    # .. your test logic from XYZ.test1 ...

def test_xyz_2(setup_data):
    # .. your test logic from XYZ.test2 ...
```

Most of our classes are not complex, so converting all to functions is better IMO.

For the second pass, we can focus on  using pytest markers (@pytest.mark.slow, @pytest.mark.regression, etc.) to categorize your tests for selective running.

Reply:
Ok then it makes sense to me